packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -169,6 +170,11 @@ export function rollupPluginAstroBuildHTML(options: PluginOptions): VitePlugin {
             const jsSource = assetImports.map((sid) => `import '${sid}';`).join('\n');
             astroPageStyleMap.set(pageStyleId, jsSource);
             assetInput.add(pageStyleId);
+
+            // preserve asset order in the order we encounter them
+            for (const assetHref of assetImports) {
+              if (!pageStyleImportOrder.includes(assetHref)) pageStyleImportOrder.push(assetHref);
```

Comment:
First, we get assets in HTML order (if an asset appears multiple times, we take the first appearance)

Reply:
Of course, this isn’t perfect! It’s very possible that some pages won’t agree on the ordering of some CSS. But because we’re bundling, there’s no perfect answer there—the only solution is to not bundle CSS at all. So while this sorting won’t solve _everyone’s_ problems, it will at least improve CSS ordering as much as we can while still bundling.